flowmachine/tests/test_subscriber_location_cluster.py

```diff
@@ -78,203 +77,171 @@ def _geo_augmented_query(self):
+    assert all(s)
+
+
+def test_first_call_day_in_first_cluster(get_dataframe):
+    """
+    Test that the first ranked call day of each subscriber is in the first cluster of each subscriber.
+    """
+    cd = CallDays("2016-01-01", "2016-01-04", level="versioned-site")
+    cd_df = get_dataframe(cd)
+
+    hartigan = HartiganCluster(cd, 50)
+    har_df = hartigan.to_geopandas()
+    cd_first = cd_df[["subscriber", "site_id", "version"]].groupby("subscriber").first()
+    har_first = (
+        har_df[["subscriber", "site_id", "version"]].groupby("subscriber").first()
+    )
+
+    joined = cd_first.join(har_first, lsuffix="_cd", rsuffix="_har")
+    s = joined.apply(
+        lambda x: (x.site_id_cd in x.site_id_har) and (x.version_cd in x.version_har),
+        axis=1,
+    )
+    assert all(s)
+
+
+def test_bigger_radius_yields_fewer_clusters(get_dataframe):
+    """
+    Test whether bigger radius yields fewer clusters per subscriber
+    """
+    radius = [1, 2, 5, 10, 50]
+    cd = CallDays("2016-01-01", "2016-01-04", level="versioned-site")
+
+    h = HartiganCluster(cd, radius[0]).get_dataframe()
+    nclusters_small_radius = h.groupby("subscriber").size()
+
+    for r in radius[1:]:
+        h = HartiganCluster(cd, r).get_dataframe()
+        nclusters_big_radius = h.groupby("subscriber").size()
+        assert all(nclusters_small_radius >= nclusters_big_radius)
+        nclusters_small_radius = nclusters_big_radius
+
+
+def test_different_call_days_format(get_dataframe):
+    """
+    Test whether we can pass different call days format such as table name, SQL query and CallDays class.
+    """
+    cd = CallDays("2016-01-01", "2016-01-04", level="versioned-site")
+    har = HartiganCluster(cd, 50).get_dataframe()
+    assert isinstance(har, pd.DataFrame)
+
+    cd.store().result()
+
+    har = HartiganCluster(Table(cd.table_name), 50).get_dataframe()
+    assert isinstance(har, pd.DataFrame)
+
+    cd_query = cd.get_query()
+    har = HartiganCluster(CustomQuery(cd_query), 50).get_dataframe()
+    assert isinstance(har, pd.DataFrame)
 
-    def test_join_returns_the_same_clusters(self):
-        """
-        Test whether joining to another table for which the start and stop time are the same yields the same clusters.
-        """
-        es = EventScore(start="2016-01-01", stop="2016-01-04", level="versioned-site")
 
-        joined = (
-            self.hartigan.join_to_cluster_components(es)
-            .to_geopandas()
-            .sort_values(["subscriber", "rank", "calldays"])
+def test_call_threshold_works(get_dataframe):
+    """
+    Test whether a call threshold above 1 limits the number of clusters.
+    """
+    cd = CallDays("2016-01-01", "2016-01-04", level="versioned-site")
+
+    hartigan = HartiganCluster(cd, 50)
+    har_df = hartigan.to_geopandas()
+    assert any(har_df.calldays == 1)
+    har_df = get_dataframe(HartiganCluster(cd, 50, call_threshold=2))
+    assert all(har_df.calldays > 1)
```

Comment:
Is this correct? It appears to be the opposite of the old assertion.

Reply:
So it does.. Looking at the original test tho, it is actually broken - the second assert isn't about the Dataframe produced in the test.